src/layout/umap.c

```diff
@@ -860,11 +865,32 @@ static igraph_error_t igraph_i_layout_umap(
     /* UMAP is sometimes used on unweighted graphs, that means distances are always zero */
     IGRAPH_CHECK(igraph_i_umap_check_distances(distances, no_of_edges));
 
-    /* Trivial graphs (0 or 1 nodes) beget trivial - but valid - layouts */
-    if (no_of_nodes <= 1) {
-        IGRAPH_CHECK(igraph_matrix_resize(layout, no_of_nodes, ndim));
-        igraph_matrix_null(layout);
-        return IGRAPH_SUCCESS;
+    if (use_seed) {
+        if((igraph_matrix_nrow(res) != no_of_nodes) || (igraph_matrix_ncol(res) != ndim)) {
+          IGRAPH_ERRORF("Seed layout should have %d points in %d dimensions, got %d points in %d dimensions.",
```

Comment:
`%d` is for `int` only, but we have `igraph_integer_t`, which maps to different types on different platforms. As a rule of thumb, you should never see `%d` in igraph code. If you do, it's likely a bug.

```suggestion
          IGRAPH_ERRORF("Seed layout should have %" IGRAPH_PRId " points in %" IGRAPH_PRId " dimensions, got %" IGRAPH_PRId " points in %" IGRAPH_PRId " dimensions.",
```

This will fix the test failure.